### PR TITLE
Use failure crate in daemon_state

### PR DESCRIPTION
**Pull request by NamsooCho**

### CODE REVIEW

File: src/toxcore/dht/daemon_state.rs [hunk 1]
```diff
@@ -4,17 +4,38 @@ When toxcore starts, it deserializes states from serialized file.
...
 use toxcore::dht::server::*;
 use toxcore::dht::packed_node::*;
 use toxcore::state_format::old::*;
 use toxcore::binary_io::*;
-use toxcore::io_tokio::*;
 use toxcore::dht::kbucket::*;
 
+/// Error that can happen when calling `deserialize_old` of DhtState.
+#[derive(Debug, Fail)]
+pub enum DeserializeOldError {
```
[the remark this answers is not in the document]
Comment (NamsooCho) [reply src/toxcore/dht/daemon_state.rs]: done

**Comment by kpp** [comment]
Would you please add more tests? https://coveralls.io/builds/20137192/source?filename=src%2Ftoxcore%2Fdht%2Fdaemon_state.rs#L19

**Comment by NamsooCho** [comment]
more test added.

File: src/toxcore/dht/daemon_state.rs [hunk 2]
```diff
@@ -113,6 +145,11 @@ mod tests {
         let serialized_len = serialized_vec.len();
         assert!(DaemonState::deserialize_old(&alice, &serialized_vec[..serialized_len - 1]).wait().is_err());
 
+        // test with serialized data corrupted
+        let mut serialized_vec = DaemonState::serialize_old(&alice);
+        serialized_vec.reverse();
+        assert!(DaemonState::deserialize_old(&alice, &serialized_vec).wait().is_err());
```
Comment (kurnevsky) [inline src/toxcore/dht/daemon_state.rs]: Do we need to check for exact error? Not just `is_err`?

File: src/toxcore/dht/daemon_state.rs [hunk 3]
```diff
@@ -113,6 +145,11 @@ mod tests {
         let serialized_len = serialized_vec.len();
         assert!(DaemonState::deserialize_old(&alice, &serialized_vec[..serialized_len - 1]).wait().is_err());
 
+        // test with serialized data corrupted
+        let mut serialized_vec = DaemonState::serialize_old(&alice);
+        serialized_vec.reverse();
```
Comment (kurnevsky) [inline src/toxcore/dht/daemon_state.rs]: Why not just define an array? It seems a bit strange that it's reversed to make it invalid.

Comment (NamsooCho) [reply src/toxcore/dht/daemon_state.rs]: done

**Pull request merged**